### PR TITLE
Push notifications: Spike for Android (and partial iOS)

### PR DESCRIPTION
**Pull request by oskarth**
Progress for push notifications spike for Android. Related to https://github.com/status-im/status-react/issues/1599 and achieves (1) "ability to manually trigger a push notification." by exposing an FCMToken.

PR only for devs to tests, nothing user-facing here. Mostly sanity check, but welcome to check actual notifications too.

### Summary:
- Print FCMToken which can be used to send PNs from Firebase Console
- Include react-native-fcm and related packages
- Update capabilities for iOS
- google service config files for Android and iOS

### Steps to test:

iOS no crash test:
- Open Status on iOS, make sure it doesn't crash. If using device, note whether it asks for permissions or not.

Android emulator test:
- Open Status on Android, if using Android ensure you have google-play-services installed. Using Genymotion this can be achieved with 2.10 and Open Gapps installed (https://github.com/opengapps/opengapps) - if you have issues PM me.
- Run `adb logcat | grep token` and note token
- Put app in background and go to Firebase Console (https://firebase.google.com/docs/cloud-messaging/) to trigger notification to that specific device.

status: ready

### CODE REVIEW

File: modules/react-native-status/android/react-native-status.iml [hunk 1]
```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yenda) [inline modules/react-native-status/android/react-native-status.iml]: this file is usually not commited

File: android/app/app.iml [hunk 2]
```diff
@@ -0,0 +1,290 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yenda) [inline android/app/app.iml]: same

File: android/StatusIm.iml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yenda) [inline android/StatusIm.iml]: same

**Review by yenda — COMMENTED** [review]
3 of the files that you committed are generated by tooling, maybe we should add them to gitignore ?

**Comment by oskarth** [comment]
@yenda done

**Review by yenda — APPROVED** [review]

File: src/status_im/ui/screens/events.cljs [hunk 4]
```diff
@@ -264,3 +266,11 @@
   :update-geolocation
   (fn [db [_ geolocation]]
     (assoc db :geolocation geolocation)))
+
+;; TODO(oskarth): Put this token in DB
+(register-handler-fx
+ :get-fcm-token
+ (fn [_ _]
+   (notifications/get-fcm-token)
+   {}))
```
Comment (alwx) [inline src/status_im/ui/screens/events.cljs:275]: hm
why does it return `{}`?

Comment (oskarth) [reply src/status_im/ui/screens/events.cljs:275]: @alwx just placeholder for effects to come - main thing is checking in progress so far, proving (1) in issue and making sure it doesn't break anything major

File: android/app/build.gradle [hunk 5]
```diff
@@ -162,6 +162,10 @@ android {
 }
 
 dependencies {
+    compile project(':react-native-udp')
```
Comment (jeluard) [inline android/app/build.gradle]: Why did this change?

File: android/app/src/main/java/im/status/ethereum/MainApplication.java [hunk 6]
```diff
@@ -43,6 +46,10 @@ public boolean getUseDeveloperSupport() {
         protected List<ReactPackage> getPackages() {
             List<ReactPackage> packages = new ArrayList<ReactPackage>(Arrays.asList(
                     new MainReactPackage(),
+                    new UdpSocketsModule(),
```
Comment (jeluard) [inline android/app/src/main/java/im/status/ethereum/MainApplication.java]: Is it related to firebase?

File: ios/StatusIm/StatusIm.entitlements [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
```
Comment (jeluard) [inline ios/StatusIm/StatusIm.entitlements:2]: What is this file?

**Review by jeluard — COMMENTED** [review]
Did you make changes to font files?

File: ios/StatusIm/Info.plist [hunk 8]
```diff
@@ -1,91 +1,97 @@
...
+      <string>SF-UI-Display-Thin.otf</string>
+      <string>Entypo.ttf</string>
+      <string>EvilIcons.ttf</string>
+      <string>FontAwesome.ttf</string>
+      <string>Foundation.ttf</string>
+      <string>Ionicons.ttf</string>
+      <string>MaterialIcons.ttf</string>
+      <string>Octicons.ttf</string>
+      <string>Zocial.ttf</string>
+      <string>MaterialCommunityIcons.ttf</string>
```
Comment (flexsurfer) [inline ios/StatusIm/Info.plist]: why do we need to add new fonts?

File: ios/StatusIm/StatusIm.entitlements (hunk 7 above)
Comment (oskarth) [reply ios/StatusIm/StatusIm.entitlements:2]: Necessary to turn on push notifications capabilities/entitlements as far as I can tell. Not present in any RN gitignore I've found. I believe it'll be fleshed out with production certificates.

Comment (jeluard) [reply ios/StatusIm/StatusIm.entitlements:2]: :+1:

File: ios/StatusIm/Info.plist (hunk 8 above)
Comment (oskarth) [reply ios/StatusIm/Info.plist]: These were autogenerated - I believe as a leftover from when I targeted API 25. Removed and seems to build fine.

File: android/app/src/main/java/im/status/ethereum/MainApplication.java (hunk 6 above)
Comment (oskarth) [reply android/app/src/main/java/im/status/ethereum/MainApplication.java]: This and TCP module imports were autogenerated - I believe as a leftover from when I targeted API 25. Removed and seems to build fine.

File: android/app/build.gradle (hunk 5 above)
Comment (oskarth) [reply android/app/build.gradle]: See below

**Review by alwx — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

**Pull request merged**